### PR TITLE
[Wait for #2166] [ Tensor ] Support NHWC for multiply and divide  @open sesame 04/07 20:08

### PR DESCRIPTION
**Pull request by jijoongmoon**
This PR includes changes of Tensor and TensorDim to support NHWC
computation for multiply and divide. It also includes unittests to
evaluate.

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

**Review by DonghakPark — APPROVED** [review]
LGTM!

File: nntrainer/tensor/tensor.cpp [hunk 1]
```diff
@@ -326,10 +334,14 @@ Tensor::Tensor(
...
 
-  for (unsigned int i = 0; i < dim.batch(); ++i)
-    for (unsigned int j = 0; j < dim.channel(); ++j)
-      for (unsigned int k = 0; k < dim.height(); ++k)
-        for (unsigned int l = 0; l < dim.width(); ++l)
+  // if fm == Tformat::NCHW, then dim[0] == batch , dim[1] == channel, dim[2] ==
+  // height, dim[3] == width. and if fm == Tformat::NHWC, dim[0] == batch,
+  // dim[1] == height, dim[2] == width, dim[3] == channel
+
+  for (unsigned int i = 0; i < dim[0]; ++i)
```
Comment (SeoHyungjun) [inline nntrainer/tensor/tensor.cpp:341]: How about using getTensorDim(idx)

File: nntrainer/tensor/tensor_dim.cpp [hunk 2]
```diff
@@ -303,8 +303,8 @@ std::vector<int> TensorDim::getEffectiveDimension(bool dynamic) const {
 bool TensorDim::is_dynamic() const { return dyn_dim_flag.any(); }
 
 std::ostream &operator<<(std::ostream &out, TensorDim const &d) {
-  out << "Shape: " << d.batch() << ":" << d.channel() << ":" << d.height()
-      << ":" << d.width() << std::endl;
+  out << "Shape: " << d[0] << ":" << d[1] << ":" << d[2] << ":" << d[3]
```
Comment (SeoHyungjun) [inline nntrainer/tensor/tensor_dim.cpp:306]: I think it can be changed as above.

**Review by SeoHyungjun — APPROVED** [review]
LGTM!

File: nntrainer/tensor/tensor.cpp [hunk 3]
```diff
@@ -720,6 +739,9 @@ Tensor Tensor::getSharedDataTensor(const TensorDim dim_, size_t offset,
                                    bool reset_stride,
                                    const std::string &name_) const {
   Tensor ret = *this;
+  if (dim_.getFormat() != ret.dim.getFormat())
```
Comment (lhs8928) [inline nntrainer/tensor/tensor.cpp:742]: How about we making a function or macro for this? Checking Tensor Format will be needed in every tensor operation.
Or at least matching the error message with line 521 might be needed.

**Comment by jijoongmoon** [comment]
This is included #2238

**Pull request closed without merging**